src/main/java/org/jenkinsci/plugins/ewm/actions/ExwsAllocateAction.java

```diff
@@ -0,0 +1,46 @@
+package org.jenkinsci.plugins.ewm.actions;
+
+import hudson.model.Action;
+import org.jenkinsci.plugins.ewm.Messages;
+import org.jenkinsci.plugins.ewm.steps.model.ExternalWorkspace;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+
+/**
+ * {@link Action} implementation for the {@link org.jenkinsci.plugins.ewm.steps.ExwsAllocateStep}.
+ * Stores the allocated {@link ExternalWorkspace}.
+ *
+ * @author Alexandru Somai
+ */
+public class ExwsAllocateAction implements Action {
```

Comment:
I would expect this action to be stored on the top level (with RunAction2 interface).
Otherwise data users will have to search for the data in the build graph, which may require extra resources